### PR TITLE
feat(code-agent): uses PR template if found

### LINKED ISSUE #1575: Code agent should follow the target repo's PR template
## Problem

When the code agent opens PRs, it uses its own freeform format instead of following the target repo's PR template (`.github/PULL_REQUEST_TEMPLATE.md`).

## Current behavior

PRs opened by the code agent have a freeform body:
- A paragraph describing the change
- A bullet list of changed files
- A `Closes #<issue>` link
- A "Post-script verification" checklist (internal to fullsend)

The repo's PR template sections are completely ignored — no "Type of change" checklist, no "How to test or reproduce" section, etc.

## Expected behavior

The agent should read `.github/PULL_REQUEST_TEMPLATE.md` from the target repo and fill in each section. Most repos have templates for a reason; ignoring them creates extra work for reviewers who need to manually request the missing information.

## Examples

- [konflux-ci/konflux-ui#925](https://github.com/konflux-ci/konflux-ui/pull/925) — doesn't follow [the repo's PR template](https://github.com/konflux-ci/konflux-ui/blob/main/.github/PULL_REQUEST_TEMPLATE.md)
- [konflux-ci/konflux-ui#865](https://github.com/konflux-ci/konflux-ui/pull/865) — same pattern, freeform body without template sections

## Implementation notes

The PR creation logic in `post-code.sh` (and the code agent definition) would need to:
1. Check for a PR template in the target repo before constructing the body
2. Parse template sections and populate them from the agent's context

Edge cases: no template (fallback to current format), multiple templates, templates with HTML comments.

**Proposed test case:**

```
Setup: Target repo with .github/PULL_REQUEST_TEMPLATE.md containing
       sections: "## Type of change", "## How to test"
Action: Code agent creates a PR
Assert: PR body contains each template section header, sections are
        populated (not left as placeholders)
```

Related: #1302 (PR body formatting — Closes keyword placement)

_Split from #1402._

### PR DESCRIPTION
**Pull request by rh-hemartin**
## Summary

Code agent now checks for PR templates during planning and structures PR descriptions to match template sections. Improves PR quality in repos with templates.

## Related Issue

Closes #1575

## Changes

- Added `pr_body` field to code-result.schema.json
- Updated post-code.sh to read `pr_body` from result file and use it as PR description
- Modified skill instructions: agent checks for `.github/pull_request_template.md` during planning, structures `pr_body` per template if found, otherwise writes best-effort description
- PR body written to result file early (during planning) so it survives agent timeouts
- No gitlint line-length constraints on `pr_body` unlike commit body

## Testing

- [ ] `make lint` passes (stage changes first, then run)
- [ ] Tests added/updated for new or modified logic

## Checklist

- [x] PR title follows [Conventional Commits](COMMITS.md) (correct type, `!` for breaking changes)
- [x] Commits are signed off (DCO) — human and human-directed agent sessions only
- [x] I wrote this contribution myself and can explain all changes in it

### CODE REVIEW

File: internal/scaffold/fullsend-repo/scripts/post-code.sh [hunk 1]
```diff
@@ -441,15 +441,28 @@ fi
...
+# Read pr_body from agent output. Fall back to commit body if absent.
+PR_BODY_FROM_RESULT=""
+if [ -n "${RESULT_FILE}" ]; then
+  PR_BODY_FROM_RESULT="$(jq -r '.pr_body // empty' "${RESULT_FILE}" 2>/dev/null || true)"
+fi
+
+if [ -n "${PR_BODY_FROM_RESULT}" ]; then
+  # Agent provided pr_body (template-aware or best-effort).
+  # Strip Signed-off-by and Closes lines so the script appends them once.
+  COMMIT_BODY="$(echo "${PR_BODY_FROM_RESULT}" | sed '/^Signed-off-by:/d' | sed '/^Closes #/d' | sed '/^Closes /d' | sed -e :a -e '/^\n*$/{ $d; N; ba; }')"
```
Comment (waynesun09) [inline internal/scaffold/fullsend-repo/scripts/post-code.sh]: **[medium]** `echo` unsafe for pr_body content — use `printf`

`echo "${PR_BODY_FROM_RESULT}"` interprets leading dashes as flags. If pr_body starts with `-e`, escape sequences (`\n`, `\t`) are silently interpreted, corrupting the PR description. With `-n`, the trailing newline is suppressed, breaking the sed pipeline.

**Suggestion:** Replace with `printf '%s\n' "${PR_BODY_FROM_RESULT}"` — `printf '%s\n'` never interprets its data argument as flags.

Same pattern in `post-code-test.sh` `build_pr_body` helper (line 139).

_Review Squad — 3-agent consensus (Claude, Gemini, Codex)_

Comment (waynesun09) [inline internal/scaffold/fullsend-repo/scripts/post-code.sh]: **[medium]** `sed '/^Closes /d'` is overly broad — strips legitimate prose

This pattern deletes ANY line starting with "Closes " — including legitimate prose like "Closes the gap between X and Y." The legacy commit-body path only strips `Closes #` (with `#`), making the new pr_body path more aggressive. Also makes the preceding `sed '/^Closes #/d'` redundant (it's subsumed by `'/^Closes /d'`).

**Suggestion:** Tighten to match only ref forms:
```bash
sed '/^Closes #/d; /^Closes [a-zA-Z0-9_.-]*\/[a-zA-Z0-9_.-]*#/d'
```
This catches `Closes #N` and `Closes org/repo#N` without false-positiving on prose.

_Review Squad — 4-agent consensus (Claude ×3, Gemini)_

File: internal/scaffold/fullsend-repo/schemas/code-result.schema.json [hunk 2]
```diff
@@ -11,6 +11,10 @@
       "type": "string",
       "description": "Branch the PR should target. Determined by the agent from issue context.",
       "pattern": "^[a-zA-Z0-9._/-]+$"
+    },
+    "pr_body": {
+      "type": "string",
+      "description": "PR description. When present, used as PR body instead of commit body. Allows markdown formatting and template-compliant structure without gitlint line-length constraints."
```
Comment (waynesun09) [inline internal/scaffold/fullsend-repo/schemas/code-result.schema.json]: **[medium]** No `maxLength` constraint on `pr_body`

The field has no length constraint, but GitHub's PR body API limit is 65,536 chars. An oversized pr_body passes schema validation but fails at `gh pr create` time, wasting a full pipeline run after the branch is already pushed.

**Suggestion:** Add `"maxLength": 65536` to catch oversized bodies at validation time:
```json
"pr_body": {
  "type": "string",
  "maxLength": 65536,
  "description": "..."
}
```

_Review Squad — 3-agent consensus (Claude ×2, Codex)_

File: internal/scaffold/fullsend-repo/skills/code-implementation/SKILL.md [hunk 3]
```diff
@@ -164,6 +164,16 @@ and `Glob` to inspect project configuration:
...
    use it when writing the commit subject in step 10.
+5. **Check for PR template.** Look for `.github/pull_request_template.md`,
+   `.github/PULL_REQUEST_TEMPLATE.md`, or `docs/pull_request_template.md`.
+   If a directory `.github/PULL_REQUEST_TEMPLATE/` exists with multiple
+   templates, use the one matching the issue context (e.g., `bug_report.md`
+   for bug fixes, `feature_request.md` for enhancements), or fall back to
+   the first/default template. If found, read it and note sections/prompts
+   — skip HTML comments (`<!-- ... -->`), extract only the visible headings
+   and prompts. You will structure the `pr_body` field in the result file
+   to match template sections (see step 6 below) — the post-script uses
```
Comment (waynesun09) [inline internal/scaffold/fullsend-repo/skills/code-implementation/SKILL.md]: **[medium]** Cross-reference "see step 6 below" points to the wrong step

Step 3, item 5 says "see step 6 below" but step 6 is "Identify the task type" (bug fix vs. feature vs. test-only) — unrelated to pr_body. The actual pr_body writing instructions are ~30 lines below, still within step 3.

**Suggestion:** Change to "see the structured output block below" or remove the parenthetical entirely.

_Review Squad — 2-agent consensus (Claude ×2)_

File: internal/scaffold/fullsend-repo/skills/code-implementation/SKILL.md [hunk 4]
```diff
@@ -186,21 +196,37 @@ use that branch. Otherwise, determine the repo's default branch:
...
+include it in pr_body. If PR template found in step 3 item 5, structure
+pr_body per template. Otherwise, write best-effort description.
 
 ```bash
 mkdir -p "${FULLSEND_OUTPUT_DIR}"
+# With PR template (match template sections):
 cat > "${FULLSEND_OUTPUT_DIR}/${FULLSEND_OUTPUT_FILE}" <<RESULT
 {
-  "target_branch": "<branch-name>"
+  "target_branch": "<branch-name>",
```
Comment (waynesun09) [inline internal/scaffold/fullsend-repo/skills/code-implementation/SKILL.md]: **[medium]** JSON escaping risk in unquoted heredoc for pr_body

The example uses unquoted `<<RESULT`, so `$` and backticks in pr_body content are shell-expanded rather than written literally. These characters are common in markdown code blocks. If the JSON is corrupted, `jq -r '.pr_body // empty'` in post-code.sh silently fails and the pr_body is discarded without warning — the agent won't know its description was lost.

**Suggestion:** Show a `jq`-based pattern as the recommended approach:
```bash
jq -n --arg tb "$branch" --arg pb "$body" \
  '{target_branch: $tb, pr_body: $pb}' \
  > "${FULLSEND_OUTPUT_DIR}/${FULLSEND_OUTPUT_FILE}"
```
Or at minimum, add a note that agents must escape `$` and backticks in pr_body values when using heredocs.

_Review Squad — 2-agent consensus (Claude ×2)_

**Review by waynesun09 — COMMENTED** [review]
**Review Squad Report** — 5 agents (Claude ×3, Gemini, Codex)

5 medium findings posted inline, all verified against the PR branch. 6 existing comments were checked — no duplicates posted.

Assisted-by: Claude (review), Gemini (review), Codex (review)

**Review by ralphbean — CHANGES_REQUESTED** [review]
This is an automated comment. We are moving agent content from `internal/scaffold/fullsend-repo/` to https://github.com/fullsend-ai/agents -- changes should be made to agent definitions there going forwards.

**Comment by rh-hemartin** [comment]
Moved to ttps://github.com/fullsend-ai/agents/pull/56
